### ISSUE TITLE
GeographicLib release build looking for debug library and can't find it

### ISSUE THREAD

**Issue by jjpepper** [microsoft/vcpkg#11484]
**Describe the bug**
I recently installed `geographiclib 1.47-patch1-10` using an up to date vcpkg.  I'm using Visual Studio 2019's CMake integration.  `find_package(GeographicLib CONFIG REQUIRED)` works and I'm able to link against the library as follows:

```target_link_libraries(miGeomExperiments PUBLIC GTest::GTest GTest::Main ${CHASM_LIB} GeographicLib GeographicLib::GeographicLib)```

I can successfully build in debug and release.  Running in debug works, but running release I get the following error:

![image](https://user-images.githubusercontent.com/10545251/82518230-76711480-9b51-11ea-93ac-52540cb359cc.png)

It's trying to load the debug version of the dll and not finding it.  

**Environment**
- OS: Windows
- Compiler: Visual Studio Enterprise 16.4.4.
- VCPKG_DEFAULT_TRIPLET=x64-windows

**To Reproduce**
1. ./vcpkg install geographiclib
2. include in package with cmake find_package(GeographicLib CONFIG REQUIRED)
3. build project in release
4. run project
5. See error (image above)

**Expected behavior**
It finds the dll and runs

**Failure logs** 
none

**Additional context**
none

**Comment by NancyLi1013** [comment]
Hi @jjpepper 
Thanks for posting this issue.
I try to build the project, but it was blocked by this.

```
CMake Error at F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake:329 (_find_package):
1> [CMake]   Could not find a configuration file for package "geographiclib" that is
1> [CMake]   compatible with requested version "".
1> [CMake] 
1> [CMake]   The following configuration files were considered but not accepted:
1> [CMake] 
1> [CMake]     F:/test/vcpkg/installed/x64-windows/share/geographiclib/geographiclib-config.cmake, version: 1.47 (package = GeographicLib, NOT geographiclib)
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeProject8/CMakeLists.txt:5 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
```
Also I noticed that you used` find_package(GeographicLib CONFIG REQUIRED)` to build.

But on my machine, It was `find_package(geographiclib CONFIG REQUIRED)`.

So I wandered if you changed something to continue the build process.

**Comment by jjpepper** [comment]
Yes I changed it to GeographicLib because that’s what the error says to do.

**Comment by jjpepper** [comment]
@NancyLi1013 it seems this issue is not happening after destroying my whole vcpkg area and install all the required packages again.  It's a bit worrying that vcpkg can get into a broken state like this, but it's likely a wider issue and not caused by this particular package, so I'd say this issue can be closed now.

**Comment by NancyLi1013** [comment]
@jjpepper 
Thanks for your deep investigation and instant update.
Please open a new issue if you have some new points about this.
